app/routes/content.js

```diff
@@ -16,16 +17,51 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, SetupSubmitContro
     },
     setup() {
         // Overrides setup method.  If query param /?edit is present, uses 'submit' controller instead.
-        if (this.get('editMode')) this.set('controllerName', 'submit');
+        this.set('controllerName', this.get('editMode') ? 'submit' : 'content');
         return this._super(...arguments);
     },
-    renderTemplate: function() {
+    renderTemplate() {
         // Overrides renderTemplate method.  If query param /?edit is present, uses 'submit' template instead.
-        return this.get('editMode') ? this.render('submit') : this._super(...arguments);
+        this.render(this.get('editMode') ? 'submit' : 'content');
     },
     model(params) {
-        if (params.edit) this.set('editMode', true);
-        return this.store.findRecord('preprint', params.preprint_id);
+        if (params.edit)
+            this.set('editMode', true);
+
+        const findPreprint = this
+            .store
+            .findRecord('preprint', params.preprint_id);
+
+        // If we're already on a branded page, stay there.
+        if (this.get('theme.isProvider'))
+            return findPreprint;
+
+        // If we're on the OSF
+        return findPreprint
```

Comment:
This redirect process from a /:preprint_id to a preprints/:provider/:preprint_id/ briefly flashes a Page Not Found`.